### PR TITLE
housekeeping: Added C# coding style to software-style-guide

### PR DESCRIPTION
**Pull request by RLittlesII**
Fixes: #166

### CODE REVIEW

File: input/contribute/software-style-guide/code-style.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+11. We use language keywords instead of BCL types (e.g. `int, string, float` instead of `Int32, String, Single`, etc) for both type references as well as method calls (e.g. `int.Parse` instead of `Int32.Parse`). See issue [391](https://github.com/dotnet/corefx/issues/391) for examples.
+12. We use PascalCasing to name all our constant local variables and fields. The only exception is for interop code where the constant value should exactly match the name and value of the code you are calling via interop.
+13. We use ```nameof(...)``` instead of ```"..."``` whenever possible and relevant.
+14. Fields should be specified at the top within type declarations.
+15. When including non-ASCII characters in the source code use Unicode escape sequences (\uXXXX) instead of literal characters. Literal non-ASCII characters occasionally get garbled by a tool or editor.
+16. When using labels (for goto), indent the label one less than the current indentation.
+
+We have provided an analyzer ruleset (`reactiveui.ruleset`) at the root of the reactiveui repository, enabling C# auto-formatting conforming to the above guidelines.
```
Comment (glennawatson) [inline input/contribute/software-style-guide/code-style.md]: Changed to analyzers.ruleset so we can more easily copy it around to other projects.

Comment (RLittlesII) [reply input/contribute/software-style-guide/code-style.md]: Resolved

File: input/contribute/software-style-guide/code-style.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+        public event NotifyCollectionChangedEventHandler CollectionChanged;
+
+        public int Count
+
+### Example File:
+
+``ObservableLinkedList`1.cs:``
+
+```C#
```
Comment (worldbeater) [inline input/contribute/software-style-guide/code-style.md]: Something went wrong with markup here :smile_cat:

Comment (RLittlesII) [reply input/contribute/software-style-guide/code-style.md]: This should be resolved.

**Pull request merged**